core/src/main/scala/tech/beshu/ror/accesscontrol/blocks/rules/FilterRule.scala

```diff
@@ -44,14 +43,25 @@ class FilterRule(val settings: Settings)
         case Left(_: Unresolvable) =>
           Rejected()
         case Right(filter) =>
-          Fulfilled(blockContext.withAddedContextHeader(Header(Name.transientFilter, filter)))
+          blockContext match {
+            case bc: SearchRequestBlockContext => Fulfilled(addFilter(bc: SearchRequestBlockContext, filter).asInstanceOf[B])
```

Comment:
this is why we should pattern match by `BlockContextUpdater[B]` instead (to avoid asInstanceOf and take advantage of exhausted pattern matching). But we can improve it in next (fields) change